apps/api/src/app/workflows-v1/usecases/get-notification-template/get-notification-template.usecase.ts

```diff
@@ -9,29 +10,18 @@ import { GetNotificationTemplateCommand } from './get-notification-template.comm
  */
 @Injectable()
 export class GetNotificationTemplate {
-  constructor(private notificationTemplateRepository: NotificationTemplateRepository) {}
+  constructor(private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase) {}
 
   async execute(command: GetNotificationTemplateCommand): Promise<NotificationTemplateEntity> {
-    const isInternalId = NotificationTemplateRepository.isInternalId(command.workflowIdOrIdentifier);
+    const workflow = await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        identifierOrInternalId: command.workflowIdOrIdentifier,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        userId: command.userId,
+      })
+    );
```

Comment:
Use the updated `GetWorkflowByIdsUseCase` which handles all preference fetching.